server-timing/defaults.php

```diff
@@ -207,3 +207,84 @@ function( $acc, $query ) {
 	}
 }
 add_action( 'wp_loaded', 'perflab_register_default_server_timing_template_metrics' );
+
+/**
+ * Registers additional Server-Timing metrics as configured in the setting.
+ *
+ * These metrics should be registered as soon as possible. They can be added
+ * and modified in the "Tools > Server-Timing" screen.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_additional_server_timing_metrics_from_setting() {
+	$options = (array) get_option( PERFLAB_SERVER_TIMING_SETTING, array() );
+
+	if ( isset( $options['benchmarking_actions'] ) ) {
+		foreach ( $options['benchmarking_actions'] as $action ) {
+			$metric_slug = 'action-' . $action;
+
+			$measure_callback = function( $metric ) use ( $action ) {
+				$metric->measure_before();
+				add_action( $action, array( $metric, 'measure_after' ), PHP_INT_MAX, 0 );
+			};
+
+			add_action(
+				$action,
+				static function() use ( $metric_slug, $measure_callback ) {
+					perflab_server_timing_register_metric(
+						$metric_slug,
+						array(
+							'measure_callback' => $measure_callback,
+							'access_cap'       => 'exist',
+						)
+					);
+				},
+				defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : -9999
+			);
+		}
+	}
+
+	if ( isset( $options['benchmarking_filters'] ) ) {
+		foreach ( $options['benchmarking_filters'] as $filter ) {
+			$metric_slug = 'filter-' . $filter;
+
+			$measure_callback = function( $metric ) use ( $filter ) {
+				$metric->measure_before();
+				add_filter(
+					$filter,
+					static function( $passthrough ) use ( $metric ) {
+						$metric->measure_after();
+						return $passthrough;
+					},
+					PHP_INT_MAX
+				);
+			};
+
+			add_filter(
+				$filter,
+				static function( $passthrough ) use ( $metric_slug, $measure_callback ) {
+					perflab_server_timing_register_metric(
+						$metric_slug,
+						array(
+							'measure_callback' => $measure_callback,
+							'access_cap'       => 'exist',
+						)
+					);
+					return $passthrough;
+				},
+				defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : -9999
```

Comment:
```suggestion
				defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : -9223372036854775808
```

Should this be like this?

Reply:
Also a comment saying this can be removed once we change to only support PHP 7.0+

Reply:
I don't think the min integer should be hard coded since it can differ from system to system (32-bit vs 64-bit).

Actually, this can be achieved with `~PHP_INT_MAX` which is the same as `PHP_INT_MIN`:

```suggestion
				defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : ~PHP_INT_MAX
```

Reply:
I've updated this to go with `~PHP_INT_MAX` which in fact I even found used like that in one place in WP core.